Source/Concurrency/CivlCoreTypes.cs

```diff
@@ -79,12 +79,14 @@ public abstract class Action
     public HashSet<Variable> gateUsedGlobalVars;
     public HashSet<Variable> actionUsedGlobalVars;
     public HashSet<Variable> modifiedGlobalVars;
+    public ConcurrencyOptions options;
```

Comment:
I think I'd prefer if we don't keep references to options in classes like `Action`, but instead pass options in wherever necessary. Would that be less convenient?

Reply:
Ok, I will make an attempt.